platform/security/common/src/test/java/org/codice/ddf/security/common/SPMetadataParserTest.java

```diff
@@ -0,0 +1,194 @@
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class SPMetadataParserTest {
+  private static List<String> spMetadata;
+
+  private static Set<Binding> bindingSet;
+
+  private static final Map EMPTY_MAP = Collections.emptyMap();
+
+  private static String entityID2;
+  private static EntityInformation testInfo2;
+
+  private static String entityID;
+  private static EntityInformation testInfo;
+
+  private static final String FILE_PATH =
+      Paths.get("src", "test", "resources", "SPMetadata").toString();
+  private static final String FILE_PATH_2 = FILE_PATH + "2";
+  private static final String CERTIFICATE_PATH =
+      Paths.get("src", "test", "resources", "Certificate.cert").toString();
+
+  private static final String ENTITY_ID = "https://localhost:8993/services/saml/test";
+  private static final String ENTITY_ID_2 =
+      "http://localhost:80/spring-security-saml2-sample/saml/metadata";
+
+  private static final String REDIRECT_LOGOUT_URL =
+      "https://localhost:8993/services/saml/logout/redirect";
+  private static final String POST_LOGOUT_URL = "https://localhost:8993/services/saml/logout/post";
+  private static final String REDIRECT_ACS_URL =
+      "https://localhost:8993/services/saml/sso/redirect";
+  private static final String POST_ACS_URL = "https://localhost:8993/services/saml/sso/post";
+  private static final String SOAP_ACS_URL = "https://localhost:8993/services/saml/sso/soap";
+  private static String CERTIFICATE;
+
+  @BeforeClass
+  public static void setupClass() throws Exception {
+
+    // read Certificate file into certificate
+    try (Stream<String> lines = Files.lines(Paths.get(CERTIFICATE_PATH))) {
+      CERTIFICATE = lines.collect(Collectors.joining(System.lineSeparator()));
+    }
+
+    // read SPMetadata file into spMetadata
+    spMetadata = new ArrayList<>();
+    try (Stream<String> lines = Files.lines(Paths.get(FILE_PATH))) {
+      spMetadata.add(lines.collect(Collectors.joining(System.lineSeparator())));
+    }
+    try (Stream<String> lines = Files.lines(Paths.get(FILE_PATH_2))) {
+      spMetadata.add(lines.collect(Collectors.joining(System.lineSeparator())));
+    }
+
+    // set up binding set
+    bindingSet = ImmutableSet.of(Binding.HTTP_POST, Binding.HTTP_REDIRECT, Binding.SOAP);
+
+    // set up metadata
+    Map<String, EntityInformation> testMap = SPMetadataParser.parse(spMetadata, bindingSet);
+    Iterator<Map.Entry<String, EntityInformation>> it = testMap.entrySet().iterator();
+
+    // get the second metadata
+    Map.Entry<String, EntityInformation> testEntry = it.next();
+    entityID2 = testEntry.getKey();
+    testInfo2 = testEntry.getValue();
+
+    // get the first metadata
```

Comment:
✏️  You mean the second? Since the line below is the second invocation of `it.next()`?